components/appmgt/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/common/SynapsePropertiesHandler.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2005-2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * 
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.appmgt.gateway.handlers.common;
+
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.rest.AbstractHandler;
+import org.wso2.carbon.appmgt.impl.AppMConstants;
+
+import java.util.Map;
+
+public class SynapsePropertiesHandler extends AbstractHandler {
+
+	@SuppressWarnings("unchecked")
+	public boolean handleRequest(MessageContext messageContext) {
+
+		org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext) messageContext).
+				getAxis2MessageContext();
+		Map<String, String> headers = (Map<String, String>) axis2MC.getProperty(
+				org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+		String ipWithPort = headers.get("HOST");
+		String ip = ipWithPort.substring(0, ipWithPort.indexOf(':'));
```

Comment:
- Not sure whether `(String) axis2MC.getProperty("REMOTE_ADDR")` gives the same IP address as in this code.
- Cannot use pre-defined constant `AppMConstants.HOST` here, because its value is "Host", not "HOST".

Fixed: String values are defined as constants (see commit f678cc847a7bfd7ee83c758386fe3b23d7e61b07 of PR #153) [in the same class](http://stackoverflow.com/a/66228/1577286).